server/resp/src/main/java/org/infinispan/server/resp/json/JsonDebugMemoryFunction.java

```diff
@@ -0,0 +1,85 @@
+package org.infinispan.server.resp.json;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import org.infinispan.commons.CacheException;
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.functional.EntryView.ReadWriteEntryView;
+import org.infinispan.server.resp.ExternalizerIds;
+import org.infinispan.util.function.SerializableFunction;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.jayway.jsonpath.JsonPath;
+
+public class JsonDebugMemoryFunction
+      implements SerializableFunction<ReadWriteEntryView<byte[], JsonBucket>, List<Integer>> {
+   public static final String ERR_PATH_CAN_T_BE_NULL = "path can't be null";
+   public static final AdvancedExternalizer<JsonDebugMemoryFunction> EXTERNALIZER = new JsonDebugMemoryFunction.Externalizer();
+
+   byte[] path;
+
+   public JsonDebugMemoryFunction(byte[] path) {
+      requireNonNull(path, ERR_PATH_CAN_T_BE_NULL);
+      this.path = path;
+   }
+
+   @Override
+   public List<Integer> apply(ReadWriteEntryView<byte[], JsonBucket> entryView) {
+      var doc = entryView.find().orElse(null);
+      if (doc == null) {
+        return new ArrayList<>(0);
+      }
+
+      var pathStr = new String(path, StandardCharsets.UTF_8);
+      try {
+         var rootNode = JSONUtil.objectMapper.readTree(doc.value());
+         var jpCtx = JSONUtil.parserForGet.parse(rootNode);
+         JsonPath jpath = JsonPath.compile(pathStr);
+         ArrayNode nodeList = jpCtx.read(jpath);
+         List<Integer> resList = new ArrayList<>(nodeList.size());
+         for (JsonNode jsonNode : nodeList) {
+            resList.add((2 * jsonNode.toString().getBytes().length) + 38);
```

Comment:
38 is the object overhead?
the whole json doc is represented as a single byte array object, IMO we shouldn't add it for each match. I would not count it for non root object
also I can't understand the *2 factor

Reply:
Me neither. I just googled a bit on this matter, and I proposed this, without really having the knowledge. This is why I asked @wburns because he is way more accurate than me on this matter